### PR TITLE
[fp16] suppot fp16 input in nansum

### PR DESCRIPTION
**Pull request by Liyulingyue**
### PR types

Others 
### PR changes

APIs 
### Describe

- https://github.com/PaddlePaddle/Paddle/issues/50832 case 26

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_nansum_api.py [hunk 1]
```diff
@@ -72,22 +72,60 @@ def test_static_graph(self):
...
+            with fluid.program_guard(train_program, startup_program):
+                input = fluid.data(name='input', dtype='float16', shape=[2, 4])
+                out1 = paddle.nansum(input)
+                out2 = paddle.nansum(input, axis=0)
+                out3 = paddle.nansum(input, axis=-1)
+                out4 = paddle.nansum(input, axis=1, keepdim=True)
+                place = fluid.CPUPlace()
+                if fluid.core.is_compiled_with_cuda():
+                    place = fluid.CUDAPlace(0)
+                exe = fluid.Executor(place)
```
Comment (zhangting2020) [inline python/paddle/fluid/tests/unittests/test_nansum_api.py]: 由于我们已不推荐使用fluid，建议对上述代码内容做一些替换：
- fluid.program_guard -> paddle.static.program_guard
- fluid.data -> paddle.static.data
- fluid.xxPlace() -> paddle.xxPlace()
- fluid.Executor() -> paddle.static.Executor

**Comment by Liyulingyue** [comment]
@sunzhongkai588 Needs review~

**Review by zhangting2020 — APPROVED** [review]
LGTM

**Comment by luotao1** [comment]
有中文文档修改PR么？

**Pull request merged**